stdlib/public/runtime/Casting.cpp

```diff
@@ -857,28 +858,89 @@ static bool _dynamicCastToExistential(OpaqueValue *dest,
     // srcDynamicType equals nullptr we have a cast from an existential
     // container with a class instance to AnyObject. In this case no check is
     // necessary.
```

Comment:
I'm confused by the comment above:  We may know the source is a class instance, but we still have to verify whether that class instance satisfies any protocol requirements on the destination, don't we?

Reply:
I believe in this case we know that the destination type is exactly `AnyObject`.